### DISCUSSION TITLE
does rtps example work with dds example? [13599]

### DISCUSSION THREAD

**Discussion by carlstand** [eProsima/Fast-DDS#2437] [Q&A]
Hi all,

I'm runing a rtps helloworld publisher example and a dds helloworld subscriber example, the program get exception at 

> inline` Cdr& deserialize(uint32_t& ulong_t) 

in Cdr.h,

does these two types of examples be able to work with each other? :)

**Comment by JLBuenoLopez** [comment]
Hi @carlstand,

Sorry for the late reply! First of all I would like to clarify what are you referring to with RTPS example. Fast DDS previous to release v2.0.0 provided an API known as FastRTPS or PubSub API. This API is being deprecated and the standard API according to the OMG DDS specification is now in use since the aforementioned release. Right now the repository provides examples with both APIs and you may run a HelloWorldExample publisher in FastRTPS API with a HelloWorldExample subscriber in Fast DDS API. I have just checked.

But maybe you are referring to the RTPS API directly. This API is common for the FastRTPS and Fast DDS APIs and it is another layer of the library. You can also use directly this API though it is not easy for non-expert users. Also, it could be possible to communicate publisher and subscribers using different APIs if the Topics, Data types and QoS agrees.

I hope this information is useful!
